### ISSUE TITLE
R-Type II loading sequence bug

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#129]
The following scene flickers in vAmiga: 

<img width="1275" alt="Bildschirmfoto 2019-08-14 um 08 37 39" src="https://user-images.githubusercontent.com/12561945/62999518-ff467b80-be6e-11e9-83f8-53998b182481.png">

**Comment by dirkwhoffmann** [comment]
More findings: DMA activity

<img width="841" alt="Bildschirmfoto 2019-08-14 um 08 43 19" src="https://user-images.githubusercontent.com/12561945/63000263-066e8900-be71-11e9-8d0d-05ac05317a7e.png">

Arrow marks area where the graphics error occurs. 

Probably relevant: From time to time, the scene switches to a very long Copper list that manually modifies sprite positions in each raster line: 

<img width="427" alt="Bildschirmfoto 2019-08-14 um 08 52 57" src="https://user-images.githubusercontent.com/12561945/63000353-3b7adb80-be71-11e9-9ef6-3a971bb0d085.png">

Manually modifying sprite positions hasn't been tested thoroughly. This should be done first by writing new vAmigaTS tests.

**Comment by Alessandro1970** [comment]
Now the intro works bu the game often crashes for CIA:

[CRASH.TXT](https://github.com/dirkwhoffmann/vAmiga/files/3563287/CRASH.TXT)

**Comment by dirkwhoffmann** [comment]
That's helpful. It crashes here: 

 ```
  // Run shift register with generated clock signal
    if (serCounter) {
        if ((delay & (CIASerClk2 | CIASerClk1)) == CIASerClk1) {  // Positive edge
            if (serCounter == 1) {
                assert(false); // DO WE EVER REACH HERE?
                delay |= CIASerInt0; // Trigger interrupt
            }
        }
    }
```

No I now the answer to `DO WE EVER REACH HERE?`😉

I've added this assertion, because the serial shift register is not really used inside the C64, because the registers had hardware bugs. Thus, the serial register implementation is not test very well (I took the CIAs from VirtualC64 and did some minor modifications).

**Comment by dirkwhoffmann** [comment]
The flickering issue is still there. 

TODO: Write vAmigaTS test that modifies SPRxPOS manually with the Copper.

**Comment by dirkwhoffmann** [comment]
Flickering is *not* due to the BPL1CON hires bug. It still flickers.

**Comment by dirkwhoffmann** [comment]
Findings: 

SAE with "Immediate Blitter": Flickers the same way as it does in vAmiga
SAE with "Blitter Waiting Always": Works perfectly well

vAmiga with Blitter level 2: 😳

<img width="889" alt="Bildschirmfoto 2019-10-22 um 11 51 43" src="https://user-images.githubusercontent.com/12561945/67275315-9c9edb00-f4c2-11e9-9bdb-c0bb469c1342.png">

Conclusions: 

- It's almost likely a Blitter / CPU timing issue.
- There is something fundamentally wrong in vAmiga w.r.t. Blitter / CPU bus sharing.

**Comment by dirkwhoffmann** [comment]
Fixed in v0.46.1
